File: frontend/libs/aas-designer-model/src/lib/snippet-verwendung.ts
export class SnippetVerwendung {
  datum: Date | null = null;
  snippetName: string = '';
  snippetTyp: string = '';
  snippetId: number = 0;

  static fromDto(dto: any) {
    const snippetVerwendung = new SnippetVerwendung();
    Object.assign(snippetVerwendung, dto);

    snippetVerwendung.datum = new Date(dto.datum);

    return snippetVerwendung;
  }
}

File: services/aas-designer/AasDesignerModel/Model/SnippetVerwendung.cs
namespace AasDesignerApi.Model
{
    public class SnippetVerwendung
    {
        public DateTime Datum { get; set; }
        public string SnippetName { get; set; } = string.Empty;
        public string SnippetTyp { get; set; } = string.Empty;
        public long? SnippetId { get; set; }
    }
}
